buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -183,34 +208,69 @@ synchronized void start() {
             spec.from(distroArtifact.resolve("config").toFile());
             spec.into(configFile.getParent());
         });
-        configure();
-        startElasticsearchProcess(distroArtifact);
+
+        try {
+            createWorkingDir(distroArtifact);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        createConfiguration();
+
+        plugins.forEach(plugin -> runElaticsearchBinScript(
+            "elasticsearch-plugin",
+            "install", "--batch", plugin.toString())
+        );
+
+        startElasticsearchProcess();
+    }
+
+    private void runElaticsearchBinScript(String tool, String... args) {
+        services.loggedExec(spec -> {
+            spec.setEnvironment(getESEnvironment());
+            spec.workingDir(workingDir);
+            if (OperatingSystem.current().isWindows()) {
+                spec.executable("cmd");
+                spec.args("/c");
+                spec.args("bin\\" + tool + ".bat");
+                spec.args((Object[]) args);
+            } else {
+                spec.executable("./bin/" + tool);
+                spec.args((Object[]) args);
+            }
+        });
     }
 
-    private void startElasticsearchProcess(Path distroArtifact) {
-        logger.info("Running `bin/elasticsearch` in `{}` for {}", workingDir, this);
+    private Map<String, String> getESEnvironment() {
+        Map<String, String> environment= new HashMap<>();
+        environment.put("JAVA_HOME", getJavaHome().getAbsolutePath());
+        environment.put("ES_PATH_CONF", configFile.getParent().toString());
+        environment.put("ES_JAVA_OPTS", "-Xms512m -Xmx512m");
```

Comment:
This seems like something that makes sense to be exposed as a config option rather than be hard-coded.

Reply:
There are lots of things that fall into this category.
The reason it's not a config option is that I would like  to find the use-cases that need to change these and figure out the best DSL to do so at that time. 
We will likely not do this, but just for the sake of example, we could expose something like this using templates of "small", "medium" "large" instead of direct configuration if we feel users don't care about details.